cfme/tests/test_db_migrate.py

```diff
@@ -79,159 +80,89 @@ def appliance_preupdate(temp_appliance_preconfig_funcscope_upgrade, appliance):
-        rc, out = ssh.run_command(
-            'pg_restore -v --dbname=vmdb_production {}{}'.format(loc, url_basename), timeout=600)
-        assert rc == 0, "Failed to restore new database: {}".format(out)
-        rc, out = ssh.run_rake_command("db:migrate", timeout=300)
-        assert rc == 0, "Failed to migrate new database: {}".format(out)
-        rc, out = ssh.run_rake_command(
-            'db:migrate:status 2>/dev/null | grep "^\s*down"', timeout=30)
-        assert rc != 0, "Migration failed; migrations in 'down' state found: {}".format(out)
-        # fetch GUID and REGION from the DB and use it to replace data in /var/www/miq/vmdb/GUID
-        # and /var/www/miq/vmdb/REGION respectively
-        data_query = {
-            'guid': 'select guid from miq_servers',
-            'region': 'select region from miq_regions'
-        }
-        for data_type, db_query in data_query.items():
-            data_filepath = '/var/www/miq/vmdb/{}'.format(data_type.upper())
-            rc, out = ssh.run_command(
-                'psql -d vmdb_production -t -c "{}"'.format(db_query), timeout=15)
-            assert rc == 0, "Failed to fetch {}: {}".format(data_type, out)
-            db_data = out.strip()
-            assert db_data, "No {} found in database; query '{}' returned no records".format(
-                data_type, db_query)
-            rc, out = ssh.run_command(
-                "echo -n '{}' > {}".format(db_data, data_filepath), timeout=15)
-            assert rc == 0, "Failed to replace data in {} with '{}': {}".format(
-                data_filepath, db_data, out)
-        # fetch v2_key
-        try:
-            rc, out = ssh.run_command(
-                'curl "{}"'.format(v2key_url), timeout=15)
-            assert rc == 0, "Failed to download v2_key: {}".format(out)
-            assert ":key:" in out, "Not a v2_key file: {}".format(out)
-            rc, out = ssh.run_command(
-                'curl -o "/var/www/miq/vmdb/certs/v2_key" "{}"'.format(v2key_url), timeout=15)
-            assert rc == 0, "Failed to download v2_key: {}".format(out)
-        # or change all invalid (now unavailable) passwords to 'invalid'
-        except AssertionError:
-            rc, out = ssh.run_command("fix_auth -i invalid", timeout=45)
-            assert rc == 0, "Failed to change invalid passwords: {}".format(out)
-        # fix db password
-        rc, out = ssh.run_command("fix_auth --databaseyml -i {}".format(
-            app_creds['password']), timeout=45)
-        assert rc == 0, "Failed to change invalid password: {}".format(out)
+    app.db.create()
+    # restore new DB
+    rc, out = app.ssh_client.run_command(
+        'pg_restore -v --dbname=vmdb_production {}{}'.format(loc, url_basename), timeout=600)
+    assert rc == 0, "Failed to restore new database: {}".format(out)
+    app.db.migrate()
+    # fetch v2_key
+    try:
+        rc, out = app.ssh_client.run_command(
```

Comment:
**Optional**

Same as above, but in this case you're actually using `out`

If you set the return of run_command to a single variable, it will be an SSHResult object. You can then use this object for asserting against, for checking strings in - it implements several built-in (__str__, __cmp__) methods to provide this functionality.

So you could say:
```python
result = app.ssh_client.run_command('curl some stuff')
assert result
assert ':key:' in result, 'bad mojo dude'